cmd/registry-replacer/main.go

```diff
@@ -0,0 +1,210 @@
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"regexp"
+	"strings"
+	"sync"
+
+	"github.com/sirupsen/logrus"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/yaml"
+
+	"github.com/openshift/ci-tools/pkg/api"
+	"github.com/openshift/ci-tools/pkg/config"
+)
+
+func main() {
+	configDir := flag.String("config-dir", "", "The directory with the ci-operator configs")
+	flag.Parse()
+	if *configDir == "" {
+		logrus.Fatal("--config-dir arg is required")
+	}
+
+	var errs []error
+	errLock := &sync.Mutex{}
+	wg := sync.WaitGroup{}
+	if err := config.OperateOnCIOperatorConfigDir(
+		*configDir,
+		func(config *api.ReleaseBuildConfiguration, info *config.Info) error {
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				if err := replacer(githubFileGetterFactory, ioutil.WriteFile)(config, info); err != nil {
+					errLock.Lock()
+					errs = append(errs, err)
+					errLock.Unlock()
+				}
+			}()
+			return nil
+		},
+	); err != nil {
+		logrus.WithError(err).Fatal("Failed to operate on ci-operator-config")
+	}
+	wg.Wait()
+	if err := utilerrors.NewAggregate(errs); err != nil {
+		logrus.WithError(err).Fatal("Encountered errors")
+	}
+}
+
+// replacer ensures replace directives are in place. It fetches the files via http because using git
+// en masse easily kills a developer laptop whereas the http calls are cheap and can be parallelized without
+// bounds.
+func replacer(
+	githubFileGetterFactory func(org, repo, branch string) githubFileGetter,
+	writer func(name string, data []byte, mode os.FileMode) error,
```

Comment:
Could this just be an `io.Writer` since we know at the call-site what the name is, and the mode is static?

Reply:
No, the `io.Writer` has no notion of a filename which we need to pass

Reply:
I thought you do know the filename anyway, so you could close over it?

Reply:
Oh right, updated it